modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/state/DefaultResourceOwnerOAuthContext.java

```diff
@@ -93,4 +100,19 @@ public String getResourceOwnerId() {
   public void setRefreshUserOAuthContextLock(Lock refreshUserOAuthContextLock) {
     this.refreshUserOAuthContextLock = refreshUserOAuthContextLock;
   }
+
+  @Override
+  public DancerState getDancerState() {
+    return null;
+  }
+
+  @Override
+  public void setDancerState(DancerState dancerState) {
+    // Nothing to do
```

Comment:
why? why can't this base class have a default handling on this? This API is lying to the people